awswrangler/distributed/ray/datasources/arrow_csv_datasource.py

```diff
@@ -0,0 +1,69 @@
+"""Ray ArrowCSVDatasource Module."""
+from typing import Any, Iterator
+
+import pandas as pd
+import pyarrow as pa
+from pyarrow import csv
+from ray.data.block import BlockAccessor
+
+from awswrangler._arrow import _add_table_partitions
+from awswrangler.distributed.ray.datasources.pandas_file_based_datasource import PandasFileBasedDatasource
+
+
+class ArrowCSVDatasource(PandasFileBasedDatasource):  # pylint: disable=abstract-method
+    """CSV datasource, for reading and writing CSV files using PyArrow."""
+
+    _FILE_EXTENSION = "csv"
+
+    def _read_stream(  # type: ignore  # pylint: disable=arguments-differ
+        self,
+        f: pa.NativeFile,
+        path: str,
+        path_root: str,
+        dataset: bool,
+        **reader_args: Any,
+    ) -> Iterator[pd.DataFrame]:
+        read_options = reader_args.get("read_options", csv.ReadOptions(use_threads=False))
+        parse_options = reader_args.get(
+            "parse_options",
+            csv.ParseOptions(),
+        )
+        convert_options = reader_args.get("convert_options", csv.ConvertOptions())
+
+        reader = csv.open_csv(
+            f,
+            read_options=read_options,
+            parse_options=parse_options,
+            convert_options=convert_options,
+        )
+
+        schema = None
+        while True:
+            try:
+                batch = reader.read_next_batch()
+                table = pa.Table.from_batches([batch], schema=schema)
+                if schema is None:
+                    schema = table.schema
+
+                if dataset:
+                    table = _add_table_partitions(
+                        table=table,
+                        path=f"s3://{path}",
+                        path_root=path_root,
+                    )
+
+                yield table.to_pandas()
```

Comment:
Any reason why we are converting `to_pandas` here? Ray is not doing it: https://github.com/ray-project/ray/blob/master/python/ray/data/datasource/csv_datasource.py#L54
I think it's wiser to delegate this conversion to the `_to_modin` method as they are most likely optimizing this conversion better than us